components/scroller/scroller.vue

```diff
@@ -0,0 +1,163 @@
+      #default="{ item, index, active }"
+    >
+      <slot
+        v-bind="{
+          item: item,
+          index,
+          active,
+        }"
+      />
+    </template>
+  </component>
+</template>
+
+<script>
+import CoreScroller from './modules/core-scroller.vue';
+import DynamicScroller from './modules/dynamic-scroller.vue';
+
+export default {
+  name: 'DtScroller',
+
+  components: {
+    CoreScroller,
+    DynamicScroller,
+  },
+
+  provide () {
+    // use function syntax so that we can access `this`
+    return {
+      emit: this.$emit,
+    };
+  },
+
+  inheritAttrs: false,
+
+  props: {
+    /**
+      * The items to render.
+      * If the items are simple arrays, the index will be used as the key.
+      * If the items are objects, the keyField will be used as the key.
+     */
+    items: {
+      type: Array,
+      required: true,
+    },
+
+    /**
+      * The width of the scroller.
+      * Can be a number (in pixels) or a string (in CSS units).
+     */
+    scrollerWidth: {
+      type: [String, Number],
+      default: '100%',
+    },
+
+    /**
+      * The height of the scroller.
+      * Can be a number (in pixels) or a string (in CSS units).
+     */
+    scrollerHeight: {
+      type: [String, Number],
+      default: '100%',
+    },
+
+    /**
+      * Indicates if the items are dynamic.
+      * If true, the items will be wrapped in a DtScrollerItem component.
+      * This is required for dynamic items to be able to react to changes in their size.
+     */
+    dynamic: {
+      type: Boolean,
+      default: false,
+    },
+
+    /**
+      * The key field to use for the items.
+      * If the items are objects, the scroller needs to be able to identify them.
+      * By default it will look for an id field on the items.
+      * This can be configured with this prop if you are using another field name.
+     */
+    keyField: {
+      type: String,
+      default: 'id',
+    },
+
+    /**
+      * The direction of the scroller.
+      * Can be either 'vertical' or 'horizontal'.
+     */
+    direction: {
+      type: String,
+      default: 'vertical',
+      validator: (value) => ['vertical', 'horizontal'].includes(value),
+    },
+
+    /**
+      * The tag to use for the list.
+     */
+    listTag: {
+      type: String,
+      default: 'div',
+    },
+
+    /**
+      * The tag to use for the items.
+     */
+    itemTag: {
+      type: String,
+      default: 'div',
+    },
+
+    /**
+      * Display height (or width in horizontal mode) of the items in pixels
+      * used to calculate the scroll size and position.
+     */
+    itemSize: {
+      type: Number,
+      default: null,
+    },
+
+    /**
+      * Minimum size used if the height (or width in horizontal mode) of a item is unknown.
+      * Is required for the initial render of items in DYNAMIC size mode.
+     */
+    minItemSize: {
+      type: [Number, String],
+    },
+  },
+
+  emits: ['scroll-start', 'scroll-end'],
```

Comment:
Mhmm....
This emits are fired by scroller.vue, Im injecting it in core-scroller (its fired from core-scroller but the emit its in scroller.vue)

Previously it was fired from core-scroller, but I updated it on this way to expose it in OUR component and because it was more easily to test it.

You can check it if you listen it in scroller_default
Also on Vue tools

![image](https://user-images.githubusercontent.com/89984179/224851449-03bfc385-a247-47d3-9d22-8096842c9fd8.png)

Reply:
Or maybe I'm not understanding correct what you mean.

Reply:
Just that it's not triggering an event in storybook, I will check this out.

Reply:
See https://github.com/dialpad/dialtone-vue/pull/812

Reply:
Great.
It was necessary to put on the story
`@scroll-start="$attrs.onScrollStart"
@scroll-end="$attrs.onScrollEnd"`

To work properly

Reply:
We should still remove the provide emit thing we've got going on here. That's not a normal way to emit an event.

Reply:
Yes, now the test works with this addition on scroller.vue
`
@scroll-start="$emit('scroll-start')"
@scroll-end="$emit('scroll-end')"
`

Reply:
nice, looks good.